katana/grib_crop_wgrib2.py

```diff
@@ -11,39 +11,55 @@
 fmt2 = '%H'
 
 
+def wind_ninja_output_dir(out_dir, file_datetime):
+    """Create the wind ninja output directory for HRRR files
+
+    Arguments:
+        out_dir {str} -- output directory path
+        file_datetime {datetime} -- datetime of the file
+
+    Returns:
+        str -- path to the cropped HRRR data
+    """
+    return os.path.join(out_dir,
+                        'data{}'.format(file_datetime.strftime(fmt1)),
+                        'wind_ninja_data',
+                        'hrrr.{}'.format(file_datetime.strftime(fmt1)))
+
+
 def call_wgrib2(action, logger):
+    """Execute a wgrib2 command
+
+    Arguments:
+        action {str} -- command for wgrib2 to execute
+        logger {logger} -- logger instance
+
+    Returns:
+        Boolean -- True if call succeeds
+    """
+
     # run command line using Popen
     logger.debug('Running "{}"'.format(action))
-    s = Popen(action, shell=True, stdout=PIPE,
-              stderr=PIPE, universal_newlines=True)
-
-    # stream the output of WindNinja to the logger
-    output = []
-    fatl = False
-    with s.stdout:
-        for line in iter(s.stdout.readline, ""):
-            logger.debug(line.rstrip())
-            output.append(line.rstrip())
-
-    with s.stderr:
-        # eline = s.stderr.readline()
-        for line in iter(s.stderr.readline, ""):
-            logger.warning(line.rstrip())
-            output.append(line.rstrip())
-
-            # if it failed, find a different forecast hour
-            if "FATAL" in line:
-                fatl = True
-
-    # if errors then create an exception
-    return_code = s.wait()
-    if return_code:
-        logger.warning("An error occured while running wgrib2 action")
-        # for line in output:
-        #     logger.warning(line)
-        fatl = True
-
-    return fatl
+
+    with subprocess.Popen(
+        action,
+        shell=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        universal_newlines=True
+    ) as s:
+
+        # stream the output of WindNinja to the logger
+        return_code = s.wait()
+        if return_code:
+            for line in s.stdout:
+                logger.warning(line.rstrip())
+            logger.warning("An error occured while running wgrib2 action")
+        else:
+            for line in s.stdout:
+                logger.debug(line.rstrip())
+
+        return return_code
```

Comment:
Nice! This prepares for having some future shared lib with common code.

Reply:
That's the hope!